File: app/src/main/java/com/example/sargiskh/android_training_java_kotlin/rx/java/grokking_rxjava/part_4/rx_with_retrofit/ApiInterface.java
package com.example.sargiskh.android_training_java_kotlin.rx.java.grokking_rxjava.part_4.rx_with_retrofit;


import java.util.List;

import retrofit2.http.GET;
import rx.Observable;

public interface ApiInterface {

    /*
    @GET("/photos")
    Call<List<Category>> getAllPhoto();
    */

    @GET("/photos")
    Observable<List<Category>> getAllPhoto();

}

File: app/src/main/java/com/example/sargiskh/android_training_java_kotlin/retrofit/kotlin/ApiInterface.kt
package com.example.sargiskh.android_training_java_kotlin.retrofit.kotlin

import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

interface ApiInterface {

    @GET("/photos")
    abstract fun getAllPhoto(): Call<List<Category>>

    companion object Factory {
        val BASE_URL = "https://jsonplaceholder.typicode.com"
        fun create(): ApiInterface {
            val retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
            return retrofit.create(ApiInterface::class.java)
        }
    }

}